### ISSUE TITLE
 Calendar popup automatically gets open when keyboard focus reaches over these controls.

### ISSUE THREAD

**Issue by BharathShanker08** [microsoft/fluentui#13646]
User Experience:
Keyboard dependent user will have difficulty if Calendar popup automatically gets opened. Keyboard user need to navigate through all the controls or need to dismiss the pop-up if the user is not intended to open the date picker.
There is situation like user is not intended to change the date but he has to navigate through the datepicker control. User will be confused if date pop-up suddenly pop's up and again he need to close the pop-up by pressing "Esc" key or something.

Test Environment:
OS : Windows 10
Version: 2004(OS Build 19592.1001)
Browser : Anaheim Dev (Version 83.0.478.5)
URL: https://developer.microsoft.com/en-us/fluentui#/controls/web/datepicker

Repro Steps:
Step 1: Open Anaheim Dev browser and turn on NVDA
Step 2: Open URL : https://developer.microsoft.com/en-us/fluentui#/controls/web/datepicker
Step 3: Navigate to Default DatePicker "Select a date" control using keyboard and observe the issue

Actual Result:
Calendar popup automatically opened when keyboard focus moves to "Select a date" control

Expected Result:
Calendar popup should not automatically open when keyboard focus moves to "Select a date" control. Upon hitting enter pop-up should be opened.

MAS Reference Link:
https://microsoft.sharepoint.com/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={a4bbd37d-ee11-485e-b5c8-0539f0ac94b2}

**Comment by paulgildea** [comment]
When repo-ing the issue I see inconsistent behavior.

When I first focus the `DatePicker` I get the behavior described. When I press `esc` and `tab` to the next element and then `shift+tab` back to the `DatePicker` I get the behavior described in the expected result. I have to press `enter` to open the pop.

@lorejoh12 is this expected behavior?

**Comment by BharathShanker08** [comment]
In forward navigation (tab navigation) also behavior should be same as reverse navigation (Shift + tab). Current behavior is explained below

**Current behavior :**
**Forward Navigation :** Up on hitting tab, focus is moving to datepicker and pop-up is automatically opening. (Here it should not open automatically upon hitting Enter only pop-up should be opened)
**Reverse Navigation :**  Up on hitting shift tab, focus is moving to datepicker and pop-up is opened upon hitting Enter (Valid behavior)

**Comment by lorejoh12** [comment]
partial dupe of #8080. This is by design, because some consumers asked for this functionality. If your app does not want this, the prop `disableAutoFocus` turns off this functionality. As discussed in 8080, we're considering changing the default from disableAutoFocus: false to true so that customers would have to explicitly opt-in, but haven't made that change yet.

**Comment by lorejoh12** [comment]
closing as by design, given that there's a prop that can be used to change this behavior
